anchore_engine/clients/localanchore_standalone.py

```diff
@@ -468,6 +470,25 @@ def make_staging_dirs(rootdir, use_cache_dir=None):
         except Exception as err:
             raise Exception("unable to prep staging directory - exception: " + str(err))
 
+    # XXX This highly questionable environment variable usage is the only way
+    # found to programmatically inject hintsfiles without requiring the
+    # hintsfile to exist in the image. Otherwise, it would require every
+    # permutation of a hintsfile to be an actual unique image. It leverages the
+    # fact that Anchore Engine will not try to extract the hinstfile if it has
+    # already been unpacked in the unpack directory.
+    try:
+        if os.environ.get('ANCHORE_TEST_HINTSFILE'):
+            test_hints = os.environ['ANCHORE_TEST_HINTSFILE']
+            destination = os.path.join(unpackdir, 'anchore_hints.json')
+            shutil.copyfile(test_hints, destination)
+    except Exception as err:
+        logger.debug("testing injection of hintsfile failed: %s", str(err))
+
+    # Set this env var so both scripts and modules that don't have access to
+    # these values have a reliable way of retrieving it. This is set here
+    # because this is the function that creates the directories.
+    os.environ['ANCHORE_ANALYZERS_UNPACKDIR'] = unpackdir
```

Comment:
In a previous review we concluded this could be removed and passed down from `run_anchore_analyzers` into `anchore_engine.analyzers.syft.catalog_image` directly as a parameter, yes?

Reply:
This has been removed+resolved!